docs/source/specs/openapi.json

```diff
@@ -1837,6 +1837,100 @@
           }
         }
       }
+    },
+    "/permissions/options/": {
+      "get": {
+        "tags": [
+          "Permission"
+        ],
+        "summary": "List the available options for fields of permissions for a tenant",
+        "description": "By default, options of application is returned. And could be resource_type or verb on demand.",
+        "operationId": "listPermissionOptions",
+        "parameters": [
+          {
+            "name": "field",
+            "in": "query",
+            "description": "specify which fields of permission to display",
+            "required": true,
+            "schema": {
+              "type": "string",
+              "enum": [
+                "application",
+                "resource_type",
+                "verb"
+              ],
+              "default": "application"
+            }
+          },
+          {
+            "name": "application",
```

Comment:
Curious, were these query filters for `application`, `resource_type` and `verb` requested when you talked with Robin? I feel like it's a bit strange to say, "give me the verb options, but only verb options that equal 'a'". Especially when we have those filters on `/permissions/`, but maybe I'm missing a use-case?

Reply:
What they want is returning all options, and also be able to filter for example "give me the verb options, but only resource_type that equal to 'a'" or "give me the resource_type options, but only application that equal to 'catalog'". And yea, the "give me the verb options, but only verb options that equal 'a'" is not our goal, just a side effect and we just not block users to use it.